### ISSUE TITLE
Name clash in class and its companion object

### ISSUE THREAD

**Issue by allanrenucci** [lampepfl/dotty#3364]
The following code snippet compiles with Dotty but not with Scalac
```scala
object Test {
  object Foo
}

class Test {
  class Foo
}
```
Same for:
```scala
object Test {
  class Foo
}

class Test {
  object Foo
}
```

**Comment by rsoeldner** [comment]
@allanrenucci I tried to tackle this, but 

```scala
val other = cls.owner.linkedClass.info.decl(cls.name).symbol.exists
```
is always _false_ for your test cases. I mean in the example:
```scala
object Test{
  object Foo
}

class Test {
  class Foo
}
```

I'm able to go from `class Foo` to the owner `class Test`, to the companion object `object Test` but I'm not able to verify `object Foo` is a member. Is there a different representation of _object_ and _class/trait_ ?

**Comment by allanrenucci** [comment]
Yes, it was a bit trickier than I thought. Indeed `object` have a special name. I fix it in #3368.
